src/multiselect.js

```diff
@@ -116,6 +116,23 @@ export class Multiselect {
     }
   }
 
+  /**
+   * Ignore multi-field updates within the given function.
+   * @param {function} func The function to call.
+   */
+  withoutMultiFieldUpdates(func) {
```

Comment:
Since this is just an event wrapper for a function, let's make the method name more general. I'm sure developers will find other use cases and it's not limited to `withoutMultiFieldUpdates`. What about the name `eventWrapper`?

Reply:
It's not really an event wrapper.  It's used to wrap code that sets dependent field updates.  And it is specifically designed to cause the multi-field updating to ignore those dependent field updates.  I'm not sure that I see any other use for it.  Given all that I think that the name is apt.

Reply:
Actually, this method can be used more than just for ignoring those dependent field updates in multi-field updates. We also use a similar structure when we do our multi-field update: https://github.com/mit-cml/workspace-multiselect/blob/main/src/multiselect.js#L256-L290 (saving any existing group id, set the group ID here, do our work, reset the group id in the finally clause). And I think this can be generalized and useful when someone just wants to hook some actions and make sure that these actions can be redo/undo together with the previous ones. (e.g. possibily dragging/pasting blocks) 

Forgive me if I'm too picky on the names, We have to think carefully about the name at this stage, as we want to avoid breaking changes, it's always a bad idea to do API renaming if later we find it can have a better name.

Reply:
I'm picky on names too.  

The fact that setting the group ID has the effect of ignoring the multi-field update is an implementation detail of the plugin.  If the implementation changed so that that was no longer the case we would change the `withoutMultiFieldUpdates` function (and it might not involve event group IDs at all). If users just want to cause undo/redo grouping, they would, presumably, already know to use `setGroup()`, since that is its intended use in Blockly.  That wouldn't be something that the Multiselect plugin would be expected to provide as a function.

Reply:
> If the implementation changed so that that was no longer the case we would change the withoutMultiFieldUpdates function (and it might not involve event group IDs at all)

Ah yeah, we should definitely take this into account as well, then it's good to keep this name.